src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -131,12 +124,7 @@ public IWebHostBuilder ConfigureServices(Action<WebHostBuilderContext, IServiceC
         /// </remarks>
         public IWebHostBuilder ConfigureAppConfiguration(Action<WebHostBuilderContext, IConfigurationBuilder> configureDelegate)
         {
-            if (configureDelegate == null)
-            {
-                throw new ArgumentNullException(nameof(configureDelegate));
-            }
-
-            _configureAppConfigurationBuilderDelegates.Add(configureDelegate);
+            _configureAppConfigurationBuilder += configureDelegate;
```

Comment:
Does this guarantee the same ordering as the List?

Reply:
[Yes](https://docs.microsoft.com/en-us/dotnet/api/system.multicastdelegate?redirectedfrom=MSDN&view=netframework-4.7.2#remarks)